builtin/providers/cloudstack/provider.go

```diff
@@ -1,30 +1,46 @@
 package cloudstack
 
 import (
+	"fmt"
+	"github.com/go-ini/ini"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
+	"os/user"
 )
 
 // Provider returns a terraform.ResourceProvider.
 func Provider() terraform.ResourceProvider {
+	usr, _ := user.Current()
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
+			"config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  fmt.Sprintf("%s/.cloudmonkey/config", usr.HomeDir),
+			},
+
+			"profile": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+
 			"api_url": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_URL", nil),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_URL", ""),
 			},
 
 			"api_key": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_KEY", nil),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_KEY", ""),
 			},
 
 			"secret_key": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_SECRET_KEY", nil),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_SECRET_KEY", ""),
 			},
```

Comment:
Please put the new options below the `secret_key`, as the existing `api_url`, `api_key` and `secret_key` would be the main way to use this provider, and alternatively you can point to a config file. So let's order the parameters in that same way...